packages/catalog-process/src/services/catalogService.ts

```diff
@@ -245,22 +245,31 @@ export function catalogServiceBuilder(
 
       return applyVisibilityToEService(eservice.data, authData);
     },
+
     async getEServices(
       authData: AuthData,
       filters: ApiGetEServicesFilters,
       offset: number,
       limit: number
     ): Promise<ListResult<EService>> {
-      // "Getting e-service with name = $name, ids = $eServicesIds, producers = $producersIds, states = $states, agreementStates = $agreementStates, limit = $limit, offset = $offset"
       logger.info(
         `Getting EServices with name = ${filters.name}, ids = ${filters.eservicesIds}, producers = ${filters.producersIds}, states = ${filters.states}, agreementStates = ${filters.agreementStates}, limit = ${limit}, offset = ${offset}`
       );
-      return await readModelService.getEServices(
-        authData.organizationId,
+      const eservicesList = await readModelService.getEServices(
+        authData,
         filters,
         offset,
         limit
       );
+
+      const eServicesToReturn = eservicesList.results.map((eservice) =>
+        applyVisibilityToEService(eservice, authData)
```

Comment:
The downside of this approach is that the `applyVisibilityToEService` function contains a path to throw `eServiceNotFound`, in case there are no descriptors or there is only one descriptor, which is in draft.
Theoretically, in this use case we will never get to that situation because those kind of eservices are already filtered out during the readModel query. So it works without any issues, but semantically is not optimal in my opinion, because we call a function that contains a path that is never needed in this flow.